### PR TITLE
Update image docs to use .docker/config.json

### PR DESCRIPTION
**Pull request by erictune**
Documents changes in #12717

### CODE REVIEW

File: docs/user-guide/production-pods.md [hunk 1]
```diff
@@ -178,11 +178,11 @@ kind: Secret
 metadata:
   name: myregistrykey
 data:
-  .dockercfg: eyAiaHR0cHM6Ly9pbmRleC5kb2NrZXIuaW8vdjEvIjogeyAiYXV0aCI6ICJabUZyWlhCaGMzTjNiM0prTVRJSyIsICJlbWFpbCI6ICJqZG9lQGV4YW1wbGUuY29tIiB9IH0K
-type: kubernetes.io/dockercfg
+  .dockerconfigjson: eyAiaHR0cHM6Ly9pbmRleC5kb2NrZXIuaW8vdjEvIjogeyAiYXV0aCI6ICJabUZyWlhCaGMzTjNiM0prTVRJSyIsICJlbWFpbCI6ICJqZG9lQGV4YW1wbGUuY29tIiB9IH0K
+type: kubernetes.io/.dockerconfigjson
```
Comment (deads2k) [inline docs/user-guide/production-pods.md]: typo.  There shouldn't be a dot in the type:  `kubernetes.io/dockerconfigjson`.

@derekwaynecarr illustration of why the `kubectl create secret` command should check types

**Comment by deads2k** [comment]
Just a typo.  Tag at will after updating.

**Comment by erictune** [comment]
Good catch.   fixed

**Comment by alph486** [comment]
This doc update is not live yet, correct? Just got pointed to the `dockerconfigjson` type via kubernetes slack. Wanted to make sure I'm not missing something :)

**Comment by erictune** [comment]
Correct.  This document describes a feature that requires kubelet v1.2 which has not been cut yet.

**Pull request merged**